New Entry and Terms: Java `StringBuilder` (#870)
Codecademy/docs: content/java/concepts/stringbuilder/terms/indexOf/indexOf.md

---
Title: '.indexOf()'
Description: 'Returns the index of the first occurrence of a substring in the StringBuilder.'
Subjects:
  - 'Computer Science'
Tags:
  - 'Characters'
  - 'Strings'
  - 'Methods'
CatalogContent:
  - 'learn-java'
  - 'paths/computer-science'
---

The **`.indexOf()`** method returns the index of the first occurrence of a substring in the `StringBuilder`. It returns `-1` if the substring is not found.

## Syntax

```pseudo

> myStringBuilder.indexOf(substring);

Reviewer: ```suggestion
myStringBuilder.indexOf(substring, index);
```

I would just consolidate into one \```pseudo snippet and delete the extra snippet on line # 25.

```

> The argument `substring` is the [`String`](https://www.codecademy.com/resources/docs/java/strings) being matched.
> 
> ```pseudo
> myStringBuilder.indexOf(substring, index);
> ```
> 
> If the optional `int` `index` is provided, `.indexOf()` will return the first occurrence of `substring` occurring after that index in the `StringBuilder`.

Reviewer: ```suggestion
The argument `substring` is the [`String`](https://www.codecademy.com/resources/docs/java/strings) being.

The `index` is an `int` argument that is optional. If provided, `.indexOf()` returns the first occurrence of `substring` after that index in the `StringBuilder`.
```
